### ISSUE TITLE
GFS.v16.3.11 - Enable the monitoring of NOAA-21 and GOES-18 radiances

### ISSUE THREAD

**Issue by KateFriedman-NOAA** [NOAA-EMC/global-workflow#1356]
**Description**

Upcoming data assimilation update to the GFS in operations will include the following (email from Daryl Kleist 2/23/23):

--------------------

* NOAA-21 ATMS & CrIS -- [JPSS-2 was launched ](https://www.nesdis.noaa.gov/next-generation/jpss-2-launch)a few months ago. After some initial issues, things seem to have been sorted out. Operations will be handed over to NOAA and it will be declared NOAA-21 soon-ish. We finally have test data available and/or flowing. These are "backbone sensors" that we need to get on as soon as we are ready. We are already missing CriS from S-NPP and NESDIS may turn off the dataflow from S-NPP some time down the road owing to bandwidth issues.
* GOES-18 Clear Sky Radiance -- This is one of the last remaining items that we were not able to deal with in January owing to lack of availability of test data. We are ready to go.
* Himawari-9 Clear Sky Radiance -- [Himawari-9 replaced -8 as the operational satellite on December 13, 2022](https://www.data.jma.go.jp/mscweb/en/oper/switchover.html). We have already dealt with the AMVs, but we still need to implement assimilation of the AHI clear sky-radiances.
* ASCAT-winds from Metop-c -- Metop-a was decommissioned in November of 2021. Despite the fact that metop-c has been operational for some time (we are already using radiances), we are reliant on NESDIS for the ascat winds product. The delivery of that product was delayed many times and we are finally ready to fix this current gap in our operations. This is an operational product that we would have otherwise like to have turned on long ago.

There is only one minor code change required to enable the assimilation of the NOAA-21 data. The rest of the components can be enabled with changes to the satinfo/convinfo files and an error covariance file for CrIS. We also need to get the CRTM-coefficients updated. We will also have to pass along spun-up bias correction coefficients for the new sensors. We are not enabling any new science.

@ADCollard will be the lead from EMC on this one and can answer questions regarding details.

Ideally, we would prefer to get this in prior to the official start of the NATL hurricane season.

--------------------

In parallel with obsproc update in operations to support this (update `obsproc_ver`).

**Workflow Changes**

New GSI tag, release notes, new obsproc version throughout.

**Tasks**
- [x] Create release branch ([release/gfs.v16.3.11](https://github.com/NOAA-EMC/global-workflow/tree/release/gfs.v16.3.11))
- [x] Updated crtm version in version files (`build.ver` and `run.ver`) = v2.4.0.1
- [x] Updated GSI tag = `gfsda.v16.3.10`
- [x] Updated UPP tag (with updated `crtm_ver`) = `upp_v8.3.0`
- [x] Updated ufs-weather-model tag (with updated `crtm_ver` and UPP tag) = `GFS.v16.3.1`
- [x] Complete release notes (copy prior release notes changes from what was v16.3.9 into release notes for v16.3.11): https://github.com/NOAA-EMC/global-workflow/blob/release/gfs.v16.3.10/docs/Release_Notes.gfs.v16.3.11.md (@KateFriedman-NOAA , @ADCollard)
~~- [ ] Install new obsproc package on supported platforms (also needed for `develop` branch).~~
- [x] Cut hand-off tag for CDF ([EMC-v16.3.11](https://github.com/NOAA-EMC/global-workflow/releases/tag/EMC-v16.3.11))
- [x] Submit CDF to NCO (@ADCollard)
- [x] Update `gfs_ver=v16.3.11` ~~and `obsproc_ver=v1.2`~~ in WCOSS2 GDA configuration as implementation occurs. @KateFriedman-NOAA 
- [x] Implementation into operations complete
- [x] Merge release branch into operational branch (`dev/gfs.v16`) - PR #2117
- [x] Cut version tag from operational branch (`gfs.v16.3.11`)
- [x] Release new version tag
- [x] Announce to users
- [x] Update Read-The-Docs operations status version in develop - PR #2121

**Comment by KateFriedman-NOAA** [comment]
@ilianagenkova @ADCollard Checking in on the status of the upcoming obsproc and GSI updates in ops. Is the obsproc update going in separately/first? Are we ready for me to make global-workflow GFSv16 release branches for either update? Thanks!

**Comment by ADCollard** [comment]
@KateFriedman-NOAA I still have a couple of weeks before I will be ready to supply you with a GSI tag.   We are still in the middle of testing!  Although I assume the actual contents of the tag could change if need be?

I know you have a deadline coming up.

**Comment by ilianagenkova** [comment]
@KateFriedman-NOAA , obsproc v1.2 release is also slowed down by testing/validating the data  that would come out of v1.2 
Previously, I discussed mid-April handoff to NCO and  late May implementation, but dates the sliding by a week or two. As soon as we understand the data I'll need a week to wrap up all code changes and testing (ambitious!)

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA I still have a couple of weeks before I will be ready to supply you with a GSI tag. We are still in the middle of testing! Although I assume the actual contents of the tag could change if need be?
> 
> I know you have a deadline coming up.

@ADCollard No rush or worries! Just wanted to know where things stood since I hadn't heard anything in a few weeks. Thanks for the update! :) Let me confirm with Iliana if the obsproc update is going in ahead of the GSI update and I can prep release branches somewhat to take the burden off anyone who may step in if I go out on leave earlier than anticipated.

> obsproc v1.2 release is also slowed down by testing/validating the data that would come out of v1.2
Previously, I discussed mid-April handoff to NCO and late May implementation, but dates the sliding by a week or two. As soon as we understand the data I'll need a week to wrap up all code changes and testing (ambitious!)

@ilianagenkova Okie dokie, thanks for the update! Do we expect the obsproc update to still go in before the GSI update? Seeing if I can prep the release branches from global-workflow a bit and knock out a couple tasks on the workflow side.

**Comment by ilianagenkova** [comment]
We can ask NCO to sync them, so you don't need to worry about version
mismach.

**Comment by KateFriedman-NOAA** [comment]
> We can ask NCO to sync them, so you don't need to worry about version mismach.

@ilianagenkova That would be ideal. Please let me know as soon as you have a decision on sync or separate. I will update global-workflow issues and create release branches once confirmed. Thanks! :)

**Comment by KateFriedman-NOAA** [comment]
@ADCollard @ilianagenkova Checking in to see if there is any update on the timeline for this ops upgrade. Thanks! :)

**Comment by ilianagenkova** [comment]
Timeline slipped to June, but I still believe we could coordinate/sync the gfs and obsproc updates, to ease the handling of the versions. @WalterKolczynski-NOAA, I guess you'd be taking over this, so I'm tagging you for awareness. Thanks!

**Comment by ADCollard** [comment]
Yes, sorry I am not quite ready for this yet.

We are hoping to be able to implement in June.

ADC

**Comment by KateFriedman-NOAA** [comment]
Okie dokie, thanks for the update! I am likely going to be out by the end of May now so I will prep a release branch in global-workflow that will be ready to get new tag and obsproc version and make sure Walter or whomever from global-workflow picks it up is up to speed.

**Comment by KateFriedman-NOAA** [comment]
@ilianagenkova Can you confirm what the obsproc version will be for this? Will it be `v1.2.0`? Thanks!

**Comment by KateFriedman-NOAA** [comment]
@ADCollard @ilianagenkova I have prepped the global-workflow release branch for the GFSv16.3.8 hand-off as best as I can before final details and tags become available. Please send PRs into the global-workflow `release/gfs.v16.3.8` branch to update the `docs/Release_Notes.gfs.v16.3.8.md` release notes (https://github.com/NOAA-EMC/global-workflow/blob/release/gfs.v16.3.8/docs/Release_Notes.gfs.v16.3.8.md). I already added details about version updates and a tentative GSI tag name based on how it's usually named.

Please check the checklist items at the top of this issue to confirm it's complete. Please confirm if we are getting a new crtm version with this, that will be additional work (e.g. update `crtm_ver`?) to get crtm updates on supported platforms for v16 and v17 support.

Let me know if it would help to have a meeting between us all (next week?) to go over the details of this upgrade, particularly for Walter's benefit. Please work with @WalterKolczynski-NOAA on this hand-off after the end of next week. Thanks!

**Comment by KateFriedman-NOAA** [comment]
I corrected my last comment to clarify which global-workflow branch to send updates to via PR (`release/gfs.v16.3.8` and not `dev/gfs.v16`).

**Comment by ilianagenkova** [comment]
@KateFriedman-NOAA , Yes , obsproc will be v1.2.0!

@KateFriedman-NOAA , @ADCollard , we are working on figuring out N-21 CrIS processing issue.
 Worst case scenario - would be acceptable to go ahead with everything else and ATMS, but not CrIS yet?

**Comment by KateFriedman-NOAA** [comment]
> Yes , obsproc will be v1.2.0!

Great, thanks for confirming @ilianagenkova !

> Worst case scenario - would be acceptable to go ahead with everything else and ATMS, but not CrIS yet?

From the workflow side it doesn't matter, assuming there will still be a new obsproc version and GSI tag for GFSv16.3.8. For worst case, would there be a follow-up obsproc update in ops for CrIS? If so, I'm guessing the obsproc version would change? Third digit change?

**Comment by ilianagenkova** [comment]
@KateFriedman-NOAA For the worst case scenario - adding N-21 CrIS slightly later won't change obsproc at all. It will be a satingest upgrade. So, no extra work i

**Comment by ilianagenkova** [comment]
For the worst case scenario  - if CrIS is added later than ATMS, obsproc won't need to change (only satingest), but GFS might...depending on how the calibration coefficients files are named. @ADCollard , what do you think?

**Comment by ADCollard** [comment]
@ilianagenkova I just discussed with Daryl.  Going ahead with ATMS and leaving CrIS to be assimilated possibly with the next GPSRO commercial buy would be acceptable.

**Comment by KateFriedman-NOAA** [comment]
> the next GPSRO commercial buy

@ADCollard Is that the update for November that I heard about or potentially earlier?

**Comment by ADCollard** [comment]
@KateFriedman-NOAA TBD.   There is pressure to get it in earlier, but whether we can will depend on the Hurricane Center.

**Comment by KateFriedman-NOAA** [comment]
@ilianagenkova @ADCollard Will the GDA need to continue archiving the CrIS dump files from @SudhirNadiga-NOAA after v16.3.8 goes in?

**Comment by ADCollard** [comment]
> @ilianagenkova @ADCollard Will the GDA need to continue archiving the CrIS dump files from @SudhirNadiga-NOAA after v16.3.8 goes in?

No

**Comment by KateFriedman-NOAA** [comment]
> No

Noted, thanks for confirming @ADCollard ! I will instruct @WalterKolczynski-NOAA to discontinue the GDA pickup (and archival) of both the ATMS and CrIS dump files from @SudhirNadiga-NOAA once v16.3.8 goes into ops.

**Comment by KateFriedman-NOAA** [comment]
@ADCollard @ilianagenkova Reminder to work with @WalterKolczynski-NOAA to complete the prep work and global-workflow tag for handing off v16.3.8 to NCO. Please see the checklist at the top of this issue for updated to-do's, including wrapping up the release notes that I started.

I am on leave after this Friday with planned return late August. Have good summers! Thanks! :)

**Comment by KateFriedman-NOAA** [comment]
This is now v16.3.9.

**Comment by KateFriedman-NOAA** [comment]
Note https://github.com/NOAA-EMC/global-workflow/issues/1857#issuecomment-1718021318 from @RussTreadon-NOAA in issue related to GFSv16.3.9:

>Part of this planned implementation is to turn on assimilation of NOAA-21 radiances. If this occurs, GFS v16.3.10 will need to add the lines
> 
> ```
>     atmsbufr       atms        n21         atms_n21            0.0     1     1
>     crisfsbufr     cris-fsr    n21         cris-fsr_n21        0.0     1     0
> ```
> 
> back to `exglobal_atmos_analysis.sh`.

**Comment by KateFriedman-NOAA** [comment]
This is now GFSv16.3.10

**Comment by KateFriedman-NOAA** [comment]
Related RFC in the RFC Memo from 9/15:
```
RFC 11569 - On WCOSS2, upgrade the satingest and tranjb software packages
to v4.7.0 and v1.0.4 respectively. This upgrade makes NOAA-21 CrIS
(Cross-track Infrared Sounder) and ATMS (Advanced Technology Microwave
Sounder) data available on WCOSS2 for eventual use by the GFS/GSI. New
eFire (enterprise Fire) active fire products and OMPS-TC (Ozone Mapping and
Profiler Suite - Total Column) data from the NCCF (NESDIS Common Cloud
Framework) are also being added as well as removing obsolete configuration
files of retired datasets. To be implemented on September 20, 1200Z to 1400Z.
```

**Comment by ADCollard** [comment]
@KateFriedman-NOAA NCO will install the new CRTM in the next couple of weeks.   It will be 2.4.0.3.  An extra versioning digit has been added to reflect adding new coefficient files (although this does contain a small code change to accommodate said files).

Do you forsee any problem with changing the version number in [build.ver](https://github.com/NOAA-EMC/global-workflow/blob/release/gfs.v16.3.10/versions/build.ver) from 2.4.0 to 2.4.0.3?

Is build.ver the only place this change needs to be made?

Unfortunately, I cannot properly test this before the CRTM is installed, but NCO (helpfully) want to install both the GFS and CRTM changes together.

**Comment by KateFriedman-NOAA** [comment]
@ADCollard On the workflow end, we would also need to update `versions/run.ver` for the runtime version of CRTM, in addition to `versions/build.ver`.

Additionally, the weather model and UPP use the crtm at build and runtime. Here is a list of the `crtm_ver` variable uses in those component's ops versions:

ufs-weather-model:
```
sorc/fv3gfs.fd/modulefiles/wcoss2/fv3.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/fv3gfs.fd/modulefiles/wcoss2/fv3.lua:load(pathJoin("crtm", crtm_ver))
sorc/fv3gfs.fd/modulefiles/hera.intel/fv3.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/fv3gfs.fd/modulefiles/hera.intel/fv3.lua:load(pathJoin("crtm", crtm_ver))
sorc/fv3gfs.fd/modulefiles/orion.intel/fv3.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/fv3gfs.fd/modulefiles/orion.intel/fv3.lua:load(pathJoin("crtm", crtm_ver))
```
UPP:
```
sorc/gfs_post.fd/jobs/J_NCEPPOST:export crtm_ver=${post_crtm_ver:-v2.2.6}
sorc/gfs_post.fd/jobs/J_NCEPPOST:export HOMECRTM=${HOMECRTM:-${NWROOT}/lib/crtm/${crtm_ver}}
sorc/gfs_post.fd/jobs/JGLOBAL_ATMOS_NCEPPOST:export crtm_ver=${post_crtm_ver:-v2.2.6}
sorc/gfs_post.fd/jobs/JGLOBAL_ATMOS_NCEPPOST:export HOMECRTM=${HOMECRTM:-${NWROOT}/lib/crtm/${crtm_ver}}
sorc/gfs_post.fd/modulefiles/post/post_orion.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/gfs_post.fd/modulefiles/post/post_orion.lua:load(pathJoin("crtm", crtm_ver))
sorc/gfs_post.fd/modulefiles/post/post_wcoss2.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/gfs_post.fd/modulefiles/post/post_wcoss2.lua:load(pathJoin("crtm", crtm_ver))
sorc/gfs_post.fd/modulefiles/post/post_hera.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/gfs_post.fd/modulefiles/post/post_hera.lua:load(pathJoin("crtm", crtm_ver))
sorc/gfs_post.fd/modulefiles/upp/upp_wcoss2.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/gfs_post.fd/modulefiles/upp/upp_wcoss2.lua:load(pathJoin("crtm", crtm_ver))
sorc/gfs_post.fd/modulefiles/upp/upp_hera.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/gfs_post.fd/modulefiles/upp/upp_hera.lua:load(pathJoin("crtm", crtm_ver))
sorc/gfs_post.fd/modulefiles/upp/upp_orion.lua:crtm_ver=os.getenv("crtm_ver") or "2.4.0"
sorc/gfs_post.fd/modulefiles/upp/upp_orion.lua:load(pathJoin("crtm", crtm_ver))
```

We force `crtm_ver` from the workflow level down into the components at build and runtime, so the version will make it down, but I'd want them aware since they probably want to test with this new version before it goes into ops. @junwang-noaa @WenMeng-NOAA FYI

We'll also want to fold this version update into the developmental versions of the GFS components but that isn't a rush.

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA @ADCollard  Do you know if NCO has crtm/2.4.3 pre-installed on WCOSS2? I can run the UPP standalone case to test it.

**Comment by ADCollard** [comment]
@KateFriedman-NOAA @WenMeng-NOAA There is a version of crtm 2.4.0.3 installed on Acorn (details below).  

The change mostly facilitates the inclusion of additional fix files, so I do not expect any differences for most users.   The only exception is if you are processing HIRS radiances, the changes expose an issue with the version numbers in the HIRS coefficient files.

BTW, NCO want to change the version number to 2.4.0.1

```
hang.lei@alogin02:~/eibns/nco-wcoss2> module load crtm/[2.4.0.3](http://2.4.0.3/)
hang.lei@alogin02:~/eibns/nco-wcoss2> module show crtm/[2.4.0.3](http://2.4.0.3/)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
   /lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/modulefiles/compiler/intel/19.1.3.304/crtm/2.4.0.3.lua:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
help([[]])
conflict("crtm")
setenv("crtm_ROOT","/lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/intel-19.1.3.304/crtm/[2.4.0.3](http://2.4.0.3/)")
setenv("crtm_VERSION","2.4.0.3")
setenv("CRTM_INC","/lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/intel-19.1.3.304/crtm/[2.4.0.3/include](http://2.4.0.3/include)")
setenv("CRTM_LIB","/lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/intel-19.1.3.304/crtm/[2.4.0.3/lib/libcrtm.a](http://2.4.0.3/lib/libcrtm.a)")
setenv("CRTM_FIX","/lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/intel-19.1.3.304/crtm/[2.4.0.3/fix](http://2.4.0.3/fix)")
whatis("Name: crtm")
whatis("Version: 2.4.0.3")
whatis("Category: library")
whatis("Description: crtm library")
```

**Comment by ADCollard** [comment]
@KateFriedman-NOAA @WenMeng-NOAA  Just to be clear, Steven suggested that I put in a PR to update the GFS and then test the implementations in parallel.  The new CRTM will not be implemented for a couple of weeks it seems.

**Comment by KateFriedman-NOAA** [comment]
> BTW, NCO want to change the version number to 2.4.0.1

@ADCollard So it'll be v2.4.0.1 now? Please confirm. Thanks!

> The change mostly facilitates the inclusion of additional fix files, so I do not expect any differences for most users. The only exception is if you are processing HIRS radiances, the changes expose an issue with the version numbers in the HIRS coefficient files.

@ADCollard Noted...so perhaps the only diffs will be from the analysis jobs then?

> Steven suggested that I put in a PR to update the GFS and then test the implementations in parallel.

@ADCollard Got it...we can prep the workflow changes but will need updated ufs-weather-model and UPP tags with the crtm version changes.

@WenMeng-NOAA Would you be able to cut a new UPP tag with the updated CRTM version? I can update the UPP tag in the global-workflow release branch and then when the CRTM is installed on WCOSS2 (Cactus or Dogwood) it can be run in parallel.

@junwang-noaa Similar question...can an updated ufs-weather-model tag with the new CRTM version be cut for this?

Thanks!

**Comment by ADCollard** [comment]
I did not know that new tags would be required for UPP and ufs-weather-model!   Unfortunately, NCO are asking for a new version number even when we are just adding missing coefficient files.   I hope this is not too disruptive.

@KateFriedman-NOAA I copied you on my latest e-mail to Steven concerning this.  As far as I am aware the new CRTM version number will be 2.4.0.1.

**Comment by ADCollard** [comment]
@KateFriedman-NOAA And yes, I only expect the analysis jobs to change and only for monitored (not actively assimilated) observations.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA I copied you on my latest e-mail to Steven concerning this. As far as I am aware the new CRTM version number will be 2.4.0.1.

Great, thanks and thanks for further confirming the new version.

> And yes, I only expect the analysis jobs to change and only for monitored (not actively assimilated) observations.

Excellent, thanks!

I'll start making the workflow changes and incorporate new ufs-weather-model and UPP tags once provided. I'll include the new GSI tag that @RussTreadon-NOAA cut.

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA I started an [UPP PR](https://github.com/NOAA-EMC/UPP/pull/802) for this crtm updating in UPP. Can you review it? Thanks!

**Comment by KateFriedman-NOAA** [comment]
Approved! Thanks @WenMeng-NOAA !

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA The new UPP tag "[upp_v8.3.0](https://github.com/NOAA-EMC/UPP/releases/tag/upp_v8.3.0)" was cut with harsh c5f3053.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA The new UPP tag "[upp_v8.3.0](https://github.com/NOAA-EMC/UPP/releases/tag/upp_v8.3.0)" was cut with harsh c5f3053.

Awesome, thanks @WenMeng-NOAA !

@ADCollard would you mind adding this new UPP tag to your PR? Changes would be to the UPP tag name in `sorc/checkout.sh`, `Externals.cfg`, and the release notes. Was just reminded...need to update the GSI tag in `Externals.cfg` as well please. Will see when the new ufs-weather-model tag comes in and if I'll update that in a follow-up PR or the current one. Thanks!

**Comment by ADCollard** [comment]
Thanks @WenMeng-NOAA and @KateFriedman-NOAA, I have made the requested changes

**Comment by KateFriedman-NOAA** [comment]
@ADCollard I have tagged the release branch ([EMC-v16.3.10](https://github.com/NOAA-EMC/global-workflow/releases/tag/EMC-v16.3.10)). You may proceed with submitting the CDF. Thanks!

**Comment by ADCollard** [comment]
Thank you!

**Comment by KateFriedman-NOAA** [comment]
This is now v16.3.11. The annual CO2 fix file update (#1924) will be v16.3.10.

Renamed release branch to `release/gfs.v16.3.11`.

**Comment by ADCollard** [comment]
@KateFriedman-NOAA @WenMeng-NOAA I am testing the global-workflow now that the CRTM has been installed on WCOSS2, but I am hitting an odd roadblock when checking out UPP.   I apparently do not have permission to access the submodule below.  This has never been an issue in the past.  Do you know if something has changed?  (I can get around it for now by copying from a previous installation).  Thanks.

```
Submodule 'sorc/post_gtg.fd' (https://github.com/NCAR/UPP_GTG) registered for path 'sorc/post_gtg.fd'
Cloning into '/lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow/sorc/gfs_post.fd2/sorc/post_gtg.fd'...
Username for 'https://github.com': 
```

**Comment by ADCollard** [comment]
The CRTM installation is here:

```
You can access crtm/2.4.0.1 by doing

module use /apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304
module load crtm/2.4.0.1
```

**Comment by WenMeng-NOAA** [comment]
> @KateFriedman-NOAA @WenMeng-NOAA I am testing the global-workflow now that the CRTM has been installed on WCOSS2, but I am hitting an odd roadblock when checking out UPP. I apparently do not have permission to access the submodule below. This has never been an issue in the past. Do you know if something has changed? (I can get around it for now by copying from a previous installation). Thanks.
> 
> ```
> Submodule 'sorc/post_gtg.fd' (https://github.com/NCAR/UPP_GTG) registered for path 'sorc/post_gtg.fd'
> Cloning into '/lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow/sorc/gfs_post.fd2/sorc/post_gtg.fd'...
> Username for 'https://github.com': 
> ```

@ADCollard The gtg code is at the NCAR private repository. I will ask @YaliMao-NOAA to give you the permission.

**Comment by ADCollard** [comment]
> > @KateFriedman-NOAA @WenMeng-NOAA I am testing the global-workflow now that the CRTM has been installed on WCOSS2, but I am hitting an odd roadblock when checking out UPP. I apparently do not have permission to access the submodule below. This has never been an issue in the past. Do you know if something has changed? (I can get around it for now by copying from a previous installation). Thanks.
> > ```
> > Submodule 'sorc/post_gtg.fd' (https://github.com/NCAR/UPP_GTG) registered for path 'sorc/post_gtg.fd'
> > Cloning into '/lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow/sorc/gfs_post.fd2/sorc/post_gtg.fd'...
> > Username for 'https://github.com': 
> > ```
> 
> @ADCollard The gtg code is at the NCAR private repository. I will ask @YaliMao-NOAA to give you the permission.

@WenMeng-NOAA Thank you!

**Comment by WenMeng-NOAA** [comment]
@ADCollard and @KateFriedman-NOAA I conducted the UPP standalone test with para crtm/2.4.0.1 and model history files from ops. GFS V16.3. My test indicates the crtm/2.4.0.1 upgrade is working for UPP and no change expected from simulation satellite product generation in UPP.

**Comment by KateFriedman-NOAA** [comment]
NCO SPA update: gfs.v16.3.11 is now running in para on WCOSS2 with para obsproc.v1.2. Will see if obsproc wants to use current obsproc.v1.1 instead and implement v1.2 later.

**Comment by ilianagenkova** [comment]
Please do not wait for us and go with obsproc v1.1. We are falling behind.

**Comment by KateFriedman-NOAA** [comment]
@ADCollard FYI, there is a WAFS upgrade being prepped to go into ops this month (issue #2013). That upgrade may get in before this one and take v16.3.11. Do you happen to have an updated timeline for this upgrade? Thanks!

**Comment by KateFriedman-NOAA** [comment]
Have renamed release branch to reflect tentative new version (`release/gfs.v16.3.12`). Will need to delete [EMC-v16.3.11](https://github.com/NOAA-EMC/global-workflow/releases/tag/EMC-v16.3.11) tag to remake for WAFS hand-off (issue #2013). Will also need to update version in release branch via PR and recut hand-off tag. Still awaiting confirmation of implementation schedule for this and WAFS upgrades in ops. Moving forward with WAFS before this for now.

**Comment by KateFriedman-NOAA** [comment]
NCO wants to merge this with the WAFS update --> GFSv16.3.11

**Comment by KateFriedman-NOAA** [comment]
New tag cut for ops: https://github.com/NOAA-EMC/global-workflow/releases/tag/EMC-v16.3.11

**Comment by KateFriedman-NOAA** [comment]
Update from NCO SPA Xiaoxue:
```
We pushed crtm/2.4.0.1 and upp/8.3.0 implementation to next Monday and gfs.v16.3.11 to the week of 11/20 for now.
```

**Comment by KateFriedman-NOAA** [comment]
Mention of updated `crtm/2.4.0.1` and `upp/8.3.0` libraries in November 9th RFC memo:

```
RFC 11858 - On WCOSS2, move the following libraries from the para to prod
environment:
● crtm/2.4.0.1
● upp/8.3.0
● g2c/1.7.0
● ip/4.1.0
● sp/2.4.0
● bufr/12.0.0
● fms/2023.02.01
● madis/4.5.0
These are essential for development of various models EMC is testing. To be
implemented on November 13 at 1500Z.
```

**Comment by KateFriedman-NOAA** [comment]
The new `crtm/2.4.0.1` and `upp/8.3.0` libraries are now in production.

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA  I ran the UPP standalone test with crtm/2.4.0.1 from prod location. The test looks good for me.

**Comment by KateFriedman-NOAA** [comment]
Updates from NCO SPA Xiaoxue:

```
I rechecked out and recompiled gfs.v16.3.11 with prod crtm/[2.4.0.1](http://2.4.0.1/) and upp/8.3.0 in place.
It will take effect from today's 18z para run.
```

The `upp_ver` was not updated to `8.3.0` in `build.ver`. Xiaoxue updated it for the para build. I will fix on our end.

**Comment by KateFriedman-NOAA** [comment]
Recut [EMC-v16.3.11 tag](https://github.com/NOAA-EMC/global-workflow/releases/tag/EMC-v16.3.11) after PR #2057 to update `upp_ver=8.3.0`. FYI #2013.

**Comment by KateFriedman-NOAA** [comment]
From NCO SPA: 
```
With several issues of WAFS, we are now planning to delay the gfs.v16.3.11 implementation to the week of 12/5.
```

**Comment by KateFriedman-NOAA** [comment]
Tag recut with WAFS changes removed: https://github.com/NOAA-EMC/global-workflow/releases/tag/EMC-v16.3.11

**Comment by KateFriedman-NOAA** [comment]
RFC memo from today (it still includes mention of the WAFS part that was moved to January):
```
RFC 11902 - On WCOSS2, implement the GFS v16.3.11 upgrade which includes
the following changes:
● Enable the monitoring of NOAA-21 and GOES-18 radiances in the GFS
GSI.
● GFS will produce high resolution WAFS output and stop producing
blended 1.25 deg WAFS files to meet a 2023 ICAO ( International Civil
Aviation Organization) milestone.
These changes have been coordinated with AWC and the UKMET Office. Was
to be implemented on November 21, 1430Z to 1830Z; rescheduled for December
5, 1430Z to 1830Z.
```

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA Great progress! Thanks for your coordination efforts.

**Comment by YaliMao-NOAA** [comment]
The new WAFS rescheduled date is Jan 17.
> rescheduled for December 5, 1430Z to 1830Z.

**Comment by KateFriedman-NOAA** [comment]
Update from @ADCollard:
```
Hi Andrew,

I just put gfs.v16.3.11 in production. Please watch today's 12z run at /lfs/h1/ops/prod/com/gfs/v16.3,
standard outputs will be /lfs/h1/ops/prod/output/20231204/gfs_*.

Thanks,
Xiaoxue
```

**Comment by KateFriedman-NOAA** [comment]
Implemented a day early:
```
kate.friedman@dlogin07:~> grep gfs_ver= /lfs/h1/ops/prod/output/20231204/gfs_atmos_analysis_12.o119170269 
+++ 0 + export gfs_ver=v16.3.11
+++ 0 + gfs_ver=v16.3.11
++ 0 + eval 'export gfs_ver=v16.3.11'
+++ 0 + export gfs_ver=v16.3.11
+++ 0 + gfs_ver=v16.3.11
gfs_ver=v16.3.11
0.075 + gfs_ver=v16.3.11
0.004 + gfs_ver=v16.3.11
```

**Comment by KateFriedman-NOAA** [comment]
@ADCollard We need to get the updated CRTMv2.4.0.1 on the R&D platforms. We are now using spack-stack on those platforms. @DavidHuber-NOAA do you know who we need to put in a request with to get the new CRTM in the R&D spack-stacks? The hpc-stack repo still?

**Comment by ADCollard** [comment]
@KateFriedman-NOAA I think @Hang-Lei-NOAA is coordinating it with EPIC, but I am unsure of the details on how this works.

**Comment by Hang-Lei-NOAA** [comment]
@Andrew Collard - NOAA Federal ***@***.***> I have informed
the EPIC with detailed line-by-line command instructions to install on
RDHPCS machines. I also add a sample installation on
hera: /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack-gfsv16  and ready
to assist EPIC anytime needed.

**Comment by KateFriedman-NOAA** [comment]
> I also add a sample installation on hera: /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack-gfsv16

Thanks @Hang-Lei-NOAA ! 

> I have informed the EPIC with detailed line-by-line command instructions to install on RDHPCS machines.

Thanks for this also! Is EPIC installing in the spack-stacks on the R&D machines? We just moved global-workflow to spack-stack 1.5.1. Would need it to show up here:

```
-bash-4.2$ module avail crtm

--------------- /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.1/envs/gsi-addon/install/modulefiles/intel-oneapi-mpi/2021.5.1/intel/2021.5.0 ----------------
   crtm/2.4.0 (L)
```

**Comment by Hang-Lei-NOAA** [comment]
@Kate Friedman - NOAA Federal ***@***.***> I am working with
Alex to get this into the spack-stack.

**Comment by KateFriedman-NOAA** [comment]
> I am working with Alex to get this into the spack-stack.

Perfect! Thank you both @Hang-Lei-NOAA @AlexanderRichert-NOAA !

**Comment by KateFriedman-NOAA** [comment]
GFSv16.3.11 is in operations and tasks for this issue are complete. Will follow up on CRTM v2.4.0.1 installs for development support while working on issue #1918 (Add GFS v16.3.11[12] changes to develop). Closing issue.
